### PR TITLE
Fix `implicit-function-declaration` warnings, pass 14

### PR DESCRIPTION
**Pull request by ribbanya**
Everything done except `mobj`, which I can't figure out.

### CODE REVIEW

**Comment by ribbanya** [comment]
Commits without #682:
https://github.com/doldecomp/melee/pull/686/files/43b940d6657ab3fe63d35358bec7cae0db479709..4f8d9975507c0dced1e32a4be480c1e86dc7df75

**Comment by ribbanya** [comment]
```sh
tools/issues.sh | grep '\-Wimplicit-function-declaration'
```

File: asm/dolphin/mtx/mtxvec.s [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 .section .text  # 0x80342AA8 - 0x80342B4C
 
-.global PSMTXMUltiVec
-PSMTXMUltiVec:
+.global PSMTXMultiVec
```
Comment (r-burns) [inline asm/dolphin/mtx/mtxvec.s:5]: PSMTXMUltiVec is the actual name they used, lol

File: src/sysdolphin/baselib/class.c [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
 void _hsdClassInfoInit(void);
-HSD_ClassInfo hsdClass = { _hsdClassInfoInit };
+HSD_ClassInfo hsdClass = {
+    { _hsdClassInfoInit, FLAGS_NONE, NULL, NULL, 0, 0, NULL, NULL, NULL, 0, 0 },
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+};
```
Comment (r-burns) [inline src/sysdolphin/baselib/class.c:17]:
```suggestion
HSD_ClassInfo hsdClass = { _hsdClassInfoInit };
```
Rest of class is guaranteed to be zero-initialized

File: src/sysdolphin/baselib/class.c [hunk 3]
```diff
@@ -247,12 +264,12 @@ HSD_Class* _hsdClassAlloc(HSD_ClassInfo* info)
     return mem_piece;
 }
 
-int _hsdClassInit(HSD_Class*)
+int _hsdClassInit(HSD_Class* arg0)
```
Comment (r-burns) [inline src/sysdolphin/baselib/class.c:267]:
```suggestion
int _hsdClassInit(HSD_Class*)
```
If it's unused, just leave it unnamed

File: src/sysdolphin/baselib/class.c [hunk 4]
```diff
@@ -403,7 +428,7 @@ bool hsdObjIsDescendantOf(HSD_Obj* o, HSD_ClassInfo* p)
 
 void class_set_flags(HSD_ClassInfo* class_info, s32 set, s32 reset)
 {
-    class_info->head.flags = class_info->head.flags & ~reset | set;
+    class_info->head.flags = (class_info->head.flags & ~reset) | set;
```
Comment (r-burns) [inline src/sysdolphin/baselib/class.c:431]:
```suggestion
    class_info->head.flags = class_info->head.flags & ~reset | set;
```
Redundant

File: src/sysdolphin/baselib/fog.c [hunk 5]
```diff
@@ -1,16 +1,31 @@
...
+    { FogInfoInit, FLAGS_NONE, NULL, NULL, 0, 0, NULL, NULL, NULL, 0, 0 },
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+} } };
+
+HSD_ClassInfo hsdFogAdj = {
+    { FogAdjInfoInit, FLAGS_NONE, NULL, NULL, 0, 0, NULL, NULL, NULL, 0, 0 },
```
Comment (r-burns) [inline src/sysdolphin/baselib/fog.c:22]: Again, use the default zero-initialization.

File: src/sysdolphin/baselib/jobj.c [hunk 6]
```diff
@@ -1,18 +1,33 @@
...
 #include <sysdolphin/baselib/spline.h>
 
 void JObjInfoInit(void);
-HSD_JObjInfo hsdJObj = { JObjInfoInit };
-
-static HSD_JObjInfo* default_class;
+HSD_JObjInfo hsdJObj = {
+    { { { JObjInfoInit, FLAGS_NONE, NULL, NULL, 0, 0, NULL, NULL, NULL, 0, 0 },
+        NULL,
+        NULL,
```
Comment (r-burns) [inline src/sysdolphin/baselib/jobj.c:17]: and here

File: src/sysdolphin/baselib/lobj.c [hunk 7]
```diff
@@ -1,10 +1,21 @@
...
+#include <dolphin/mtx/vec.h>
+#include <placeholder.h>
 
 void LObjInfoInit(void);
 
-HSD_LObjInfo hsdLObj = { LObjInfoInit };
+HSD_LObjInfo hsdLObj = {
+    { { { LObjInfoInit, FLAGS_NONE, NULL, NULL, 0, 0, NULL, NULL, NULL, 0, 0 },
+        NULL,
+        NULL,
```
Comment (r-burns) [inline src/sysdolphin/baselib/lobj.c:13]: ditto

File: asm/dolphin/mtx/mtxvec.s (hunk 1 above)
Comment (r-burns) [reply asm/dolphin/mtx/mtxvec.s:5]: Never mind, I don't see either spelling in a debug map, although I see it in other games' decomp repos. Probably just a misspelled name we came up with that got copy/pasted elsewhere?

File: src/sysdolphin/baselib/class.c (hunk 4 above)
Comment (r-burns) [reply src/sysdolphin/baselib/class.c:424]: Redundant, but fixes a clang warning, and is potentially helpful if you don't recall the bitwise operation precedence, so we're gonna keep it

File: src/dolphin/os/init/__ppc_eabi_init.c [hunk 8]
```diff
@@ -11,8 +11,8 @@ extern voidfunc __ctors[];
 
 static void __init_cpp(void);
 
-SECTION_INIT asm void __init_hardware(void)
-{ // clang-format off
+SECTION_INIT asm void __init_hardware(void){
+    // clang-format off
```
Comment (ribbanya) [inline src/dolphin/os/init/__ppc_eabi_init.c:15]: idk why it keeps doing this

Comment (r-burns) [reply src/dolphin/os/init/__ppc_eabi_init.c:15]: clang-format is doing this? never had that happen for me

File: src/dolphin/os/init/__ppc_eabi_init.c [hunk 9]
```diff
@@ -3,16 +3,17 @@
...
-
+extern Event __ctors[];
 static void __init_cpp(void);
 
-SECTION_INIT asm void __init_hardware(void){
-    // clang-format off
+#ifdef MWERKS_GEKKO
+
+SECTION_INIT asm void __init_hardware(void)
+{ // clang-format off
```
Comment (r-burns) [inline src/dolphin/os/init/__ppc_eabi_init.c:16]: Oh, I get it now

**Review by r-burns — APPROVED** [review]

**Pull request merged**